libuavcan/include/uavcan/util/avl_tree.hpp

```diff
@@ -0,0 +1,373 @@
+/*
+* AVL Tree implementation.
+* Copyright (C) 2019 Theodoros Ntakouris <zarkopafilis@gmail.com>
+*/
+
+#ifndef UAVCAN_UTIL_AVL_TREE_HPP_INCLUDED
+#define UAVCAN_UTIL_AVL_TREE_HPP_INCLUDED
+
+#include <cstdlib>
+#include <cassert>
+#include <cmath>
+#include <uavcan/build_config.hpp>
+#include <uavcan/util/templates.hpp>
+#include <algorithm>
+#include <functional>
+#include <uavcan/dynamic_memory.hpp>
+#include <uavcan/debug.hpp>
+
+namespace uavcan {
```

Comment:
There are many places in the code where your formatting deviates from the style guide, please fix them.